### PR TITLE
[sc-14313] Duo Catalog Template

### PR DESCRIPTION
**Pull request by arooxa**
Duo Catalog Template - Used SCIM template as reference

[[sc-14313]](https://app.shortcut.com/sgnl/story/20157/productionize-duo-as-an-sor)

### CODE REVIEW

File: duo.sgnl.yaml
Comment (r-rakshith) [inline duo.sgnl.yaml:3]: Requesting for a shiny new Duo icon
cc: @ivanfong, @jasoncarulli

File: duo.sgnl.yaml
Comment (r-rakshith) [inline duo.sgnl.yaml:78]:
```suggestion
  # https://duo.com/docs/adminapi#groups
  Group:
```

File: duo.sgnl.yaml
Comment (r-rakshith) [inline duo.sgnl.yaml:116]: We can move this closer to User entity.

File: duo.sgnl.yaml
Comment (r-rakshith) [inline duo.sgnl.yaml:136]:
```suggestion
  # https://duo.com/docs/adminapi#phones
  Phone:
```

File: duo.sgnl.yaml
Comment (r-rakshith) [inline duo.sgnl.yaml:25]: Why is the pagesize 300 everywhere?

File: duo.sgnl.yaml
Comment (r-rakshith) [inline duo.sgnl.yaml:197]: Move this closer to User entity.

File: duo.sgnl.yaml
Comment (r-rakshith) [inline duo.sgnl.yaml:150]:
```suggestion
        type: Bool
```

File: duo.sgnl.yaml
Comment (r-rakshith) [inline duo.sgnl.yaml:190]:
```suggestion
        type: Bool
```

File: duo.sgnl.yaml
Comment (r-rakshith) [inline duo.sgnl.yaml:1]: Let's also add the Endpoint entity to this

**Review by r-rakshith — CHANGES_REQUESTED** [review]
Looks good overall. A few changes required before this is ready to be merged.

File: duo.sgnl.yaml
Comment (r-rakshith) [inline duo.sgnl.yaml:88]: This duplicate key makes the YAML invalid.

To make sure that the YAML is valid and contains all the required info, let's also upload this template to 
- a YAML validator
- a SGNL Console (`dev`/`sbx`)

Comment (arooxa) [reply duo.sgnl.yaml:88]: I'm not seeing any duplicates, which key are you referring to?

File: duo.sgnl.yaml
Comment (arooxa) [reply duo.sgnl.yaml:25]: Most endpoints accept a maximum page size of 300, although it's not standardized. There is no limit on the requested page size - eg. a page size of 99999 will be accepted but only return 300 values at most

**Review by richardxgao — CHANGES_REQUESTED** [review]
A few comments on consistency, but overall looks good!

File: duo.sgnl.yaml
Comment (r-rakshith) [reply duo.sgnl.yaml:26]: Let's stick to page size 100 as a standard. 
We can document the max allowed page size and possibly add validation on the adapter end if we know the limits on page size.

File: duo.sgnl.yaml
Comment (r-rakshith) [reply duo.sgnl.yaml:132]: Sure, we can sync on this offline.

File: duo.sgnl.yaml
Comment (r-rakshith) [reply duo.sgnl.yaml:208]: I had the same question during the first pass of review. These are strings. From the response samples, it'd be stored as `"0"`.

File: duo.sgnl.yaml
Comment (arooxa) [reply duo.sgnl.yaml:102]: Yup, parent field should be the external id of the parent identity. I changed the User externalID earlier and forgot to update this, good catch!

File: duo.sgnl.yaml
Comment (arooxa) [reply duo.sgnl.yaml:322]: @r-rakshith the SCIM catalog also uses the external id to reference attributes in the entity relationships, is this expected?

Comment (richardxgao) [reply duo.sgnl.yaml:322]: If SCIM also uses this format, and it works, then I'm OK with leaving as-is. This is not a major issue.

File: duo.sgnl.yaml
Comment (arooxa) [reply duo.sgnl.yaml:135]: Verified that this works with the console. Next step is to wait for the duo adapter to be merged and test before merging this catalog right?

**Review by richardxgao — APPROVED** [review]
LGTM, but one minor comment is not addressed: https://github.com/SGNL-ai/catalog-sgnl-templates/pull/42#discussion_r1478368876. 

If you need help testing the Duo adapter let me or Rakshith know.

**Review by r-rakshith — DISMISSED** [review]
LGTM. 
Let's test this catalog on Duo adapter once it is released to `DEV` and then merge this.

File: duo.sgnl.yaml
Comment (r-rakshith) [inline duo.sgnl.yaml:87]:
```suggestion
    description: Entity representing the Duo groups each user belongs to

File: duo.sgnl.yaml
Comment (r-rakshith) [inline duo.sgnl.yaml:108]:
```suggestion
    description: Entity representing the Phone owned by the User
```

File: duo.sgnl.yaml
Comment (r-rakshith) [inline duo.sgnl.yaml:166]:
```suggestion
    description: Phone entity in Duo
```

File: duo.sgnl.yaml
Comment (r-rakshith) [inline duo.sgnl.yaml:228]:
```suggestion
    description Endpoint entity in Duo
```

**Review by r-rakshith — COMMENTED** [review]
Also, some changes to description that addresses Richards' comments.

**Review by richardxgao — APPROVED** [review]
The remaining open comment can be addressed in a separate PR.

File: duo.sgnl.yaml
Comment (arooxa) [reply duo.sgnl.yaml:135]: finished testing

**Review by r-rakshith — APPROVED** [review]
LGTM. Thanks!

File: duo.sgnl.yaml
Comment (r-rakshith) [inline duo.sgnl.yaml:3]:
```suggestion
icon: "PHN2ZyB3aWR0aD0iNjQiIGhlaWdodD0iNjQiIHZpZXdCb3g9IjAgMCA2NCA2NCIgZmlsbD0ibm9uZSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KPHJlY3Qgd2lkdGg9IjY0IiBoZWlnaHQ9IjY0IiBmaWxsPSIjNkRCRTUwIi8+CjxnIG9wYWNpdHk9IjAuODUiPgo8cGF0aCBkPSJNMjMuNDcyIDMxLjc2OEg4VjM5LjcwNEgxNS43NDRDMTkuODg4IDM5LjcwNCAyMy4yNzIgMzYuMTkyIDIzLjQ3MiAzMS43NjhaIiBmaWxsPSJ3aGl0ZSIvPgo8L2c+CjxwYXRoIGQ9Ik0xNS43NDQgMjNIOFYzMC45MzZIMjMuNDhDMjMuMjcyIDI2LjUxMiAxOS44ODggMjMgMTUuNzQ0IDIzWiIgZmlsbD0id2hpdGUiLz4KPHBhdGggZD0iTTQ4LjI2NCAyM0M0NC4xMiAyMyA0MC43MjggMjYuNTIgNDAuNTI4IDMwLjkzNkg1NS45OTJDNTUuOCAyNi41MTIgNTIuNDA4IDIzIDQ4LjI2NCAyM1oiIGZpbGw9IndoaXRlIi8+CjxnIG9wYWNpdHk9IjAuODUiPgo8cGF0aCBkPSJNNTYgMzEuNzY4SDQwLjUyOEM0MC43MjggMzYuMTkyIDQ0LjExMiAzOS43MDQgNDguMjY0IDM5LjcwNEM1Mi40MDggMzkuNzA0IDU1LjggMzYuMTkyIDU2IDMxLjc2OFoiIGZpbGw9IndoaXRlIi8+CjwvZz4KPHBhdGggZD0iTTI0LjI1NiAyM1YzMS4zNTJDMjQuMjU2IDM1LjgyNCAyNy41MTIgMzkuNDggMzEuNjE2IDM5LjY5NlYyM0gyNC4yNTZaIiBmaWxsPSJ3aGl0ZSIvPgo8ZyBvcGFjaXR5PSIwLjg1Ij4KPHBhdGggZD0iTTM5Ljc0NCAyM0gzMi4zOTJWMzkuNzA0SDM5Ljc0NFYzMS4zNTJWMjNaIiBmaWxsPSJ3aGl0ZSIvPgo8L2c+Cjwvc3ZnPg=="
```

**Review by r-rakshith — APPROVED** [review]

**Review by richardxgao — APPROVED** [review]

**Pull request merged**
